ddp-workspace/projects/ddp-sdk/src/lib/services/authentication/auth0Adapter.service.ts

```diff
@@ -69,41 +69,52 @@ export class Auth0AdapterService implements OnDestroy {
-        // TODO refactor and document differences between localregistration flow and "normal" flow
-        let audience = 'https://' + this.configuration.auth0Domain;
-        if (!_.isUndefined(this.configuration.auth0Audience)) {
-            // when using a custom auth0 domain, the audience and domain will differ
-            // and the audience needs to be the original name of the tenant
-            audience = 'https://' + this.configuration.auth0Audience;
+        try {
+            // TODO Clarify how audience should be used, what is the correct value,etc.
+            // TODO refactor and document differences between localregistration flow and "normal" flow
+            let audience = 'https://' + this.configuration.auth0Domain;
+            if (!_.isUndefined(this.configuration.auth0Audience)) {
+                // when using a custom auth0 domain, the audience and domain will differ
+                // and the audience needs to be the original name of the tenant
+                audience = 'https://' + this.configuration.auth0Audience;
+            }
+            audience = audience + '/userinfo';
+
+            return new auth0.WebAuth({
+                domain: this.configuration.auth0Domain,
+                clientID: clientId,
+                responseType: 'token id_token',
+                audience,
+                scope: 'openid profile',
+                redirectUri
+            });
+        } catch (e) {
+            console.error(e);
+            throw new Error(e);
         }
-        audience = audience + '/userinfo';
-        return new auth0.WebAuth({
-            domain: this.configuration.auth0Domain,
-            clientID: clientId,
-            responseType: 'token id_token',
-            audience,
-            scope: 'openid profile',
-            redirectUri
-        });
     }
 
     private createLocalWebAuth(clientId: string): any {
-        // TODO see audience in createWebAuth()
-        let audience = 'https://' + this.configuration.auth0Domain;
-        if (!_.isUndefined(this.configuration.auth0Audience)) {
-            audience = 'https://' + this.configuration.auth0Audience;
+        try {
+            // TODO see audience in createWebAuth()
+            let audience = 'https://' + this.configuration.auth0Domain;
+            if (!_.isUndefined(this.configuration.auth0Audience)) {
+                audience = 'https://' + this.configuration.auth0Audience;
+            }
+            audience = audience + '/userinfo';
+            return new auth0.WebAuth({
+                domain: this.configuration.auth0Domain,
+                clientID: clientId,
+                studyGuid: this.configuration.studyGuid,
+                responseType: 'code',
+                audience,
+                scope: 'offline_access openid profile',
+                redirectUri: this.configuration.auth0CodeRedirect
+            });
+        } catch (e) {
+            console.error(e);
+            throw new Error(e);
```

Comment:
Why was this necessary?

Reply:
it wasn't but while I was working on i18n I faced again that old error about absence of adminClientId. The system throws  just a strange message about http interceptors in the console and that's it. Remember how we spent almost all day (or two days?) for pancan in the past and you realized it's related to adminClientId absence eventually and we agreed it would be better to improve the error logging in the future. So I decided to do it.

I can do it in a separate PR

Reply:
created separate PR https://github.com/broadinstitute/ddp-angular/pull/966